.github/workflows/csound_builds.yml

```diff
@@ -1,65 +1,84 @@
-        
-  emscripten_build:
-    runs-on: ubuntu-latest
-    steps:
-      - name: run the Docker image
-        run: docker run --name csound_test_run -i sjamble/csound_emscripten
-      - name: copying build out of container
-        run: docker cp csound_test_run:/csound/Emscripten/csound-web-6.17.0.zip .
-      - uses: actions/upload-artifact@v2
-        with:
-          name: csound-web-6.17.0.zip
-          path: .
-  android_build:
-    runs-on: ubuntu-latest
-    steps:
-      - name: run the Docker image
-        run: docker run --name csound_test_run -i sjamble/csound_android
-      - name: copying build out of container
-        run: docker cp csound_test_run:/csound/Android/csound-android-6.17.0.zip .
-      - uses: actions/upload-artifact@v2
-        with:
-          name: csound-android-6.17.0.zip 
-          path: .
+    macos_build:
+        runs-on: macos-latest
+        steps:
+            - name: Checkout Source Code
+              uses: actions/checkout@v1
+              with:
+                  fetch-depth: 1
+            - name: Install Dependencies
+              run: brew install cmake swig eigen asio libsndfile portaudio portmidi libsamplerate liblo jack openjdk hdf5 gettext fluid-synth fltk stk wiiuse
+            - name: Configure build
+              run: mkdir build && cd build && cmake ..
+            - name: Build Csound
+              run: cd build && make
+            - name: Run tests
+              run: cd build && make test csdtests
+
+    emscripten_build:
+        runs-on: ubuntu-latest
+        steps:
+            - name: run the Docker image
+              run: docker run --name csound_test_run -i sjamble/csound_emscripten
+            - name: copying build out of container
+              run: docker cp csound_test_run:/csound/Emscripten/csound-web-6.17.0.zip .
+            - uses: actions/upload-artifact@v2
+              with:
+                  name: csound-web-6.17.0.zip
+                  path: .
+
+    android_build:
+        runs-on: ubuntu-latest
+        steps:
+            - name: run the Docker image
+              run: docker run --name csound_test_run -i sjamble/csound_android
+            - name: copying build out of container
+              run: docker cp csound_test_run:/csound/Android/csound-android-6.17.0.zip .
+            - uses: actions/upload-artifact@v2
+              with:
+                  name: csound-android-6.17.0.zip
+                  path: .
+
+    windows_build:
+        runs-on: windows-latest
+        steps:
+            - name: Checkout Source Code
+              uses: actions/checkout@v1
+              with:
+                  fetch-depth: 1
+            - name: Install system dependencies
+              run: |
+                  choco install swig -y --version=4.0.1 --allow-downgrade
+                  choco install winflexbison -y
+            - name: Configure build
+              run: cmake -B build -S . -DVCPKG_TARGET_TRIPLET=x64-windows-csound
```

Comment:
Will the TRIPLET flag be required for anyone building on Windows?

Reply:
Yeah it will be required. This is also temporary. I will try to deduce in CMake what it should be but it was difficult as some env vars aren't set when this needs to be configured (before first project() cmake call). 

There should be a programmatic solution to pick a sane default and be allowed to be overriden via command line or env var.